op-e2e/system/fees/fees_test.go

```diff
@@ -134,26 +155,24 @@ func testFees(t *testing.T, cfg e2esys.SystemConfig) {
 
 	require.Equal(t, decimals.Uint64(), uint64(6), "wrong gpo decimals")
 
-	// BaseFee Recipient
-	baseFeeRecipientStartBalance, err := l2Seq.BalanceAt(context.Background(), predeploys.BaseFeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
-	require.Nil(t, err)
-
-	// L1Fee Recipient
-	l1FeeRecipientStartBalance, err := l2Seq.BalanceAt(context.Background(), predeploys.L1FeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
-	require.Nil(t, err)
+	baseFeeRecipientStartBalance := balanceAt(predeploys.BaseFeeVaultAddr)
+	l1FeeRecipientStartBalance := balanceAt(predeploys.L1FeeVaultAddr)
+	sequencerFeeVaultStartBalance := balanceAt(predeploys.SequencerFeeVaultAddr)
+	operatorFeeVaultStartBalance := balanceAt(predeploys.OperatorFeeVaultAddr)
 
-	sequencerFeeVaultStartBalance, err := l2Seq.BalanceAt(context.Background(), predeploys.SequencerFeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
-	require.Nil(t, err)
+	require.Equal(t, baseFeeRecipientStartBalance.Sign(), 0, "base fee vault should be empty")
+	require.Equal(t, l1FeeRecipientStartBalance.Sign(), 0, "l1 fee vault should be empty")
+	require.Equal(t, sequencerFeeVaultStartBalance.Sign(), 0, "sequencer fee vault should be empty")
+	require.Equal(t, operatorFeeVaultStartBalance.Sign(), 0, "operator fee vault should be empty")
 
 	genesisBlock, err := l2Seq.BlockByNumber(context.Background(), big.NewInt(rpc.EarliestBlockNumber.Int64()))
 	require.NoError(t, err)
 
-	coinbaseStartBalance, err := l2Seq.BalanceAt(context.Background(), genesisBlock.Coinbase(), big.NewInt(rpc.EarliestBlockNumber.Int64()))
-	require.NoError(t, err)
+	coinbaseStartBalance := balanceAt(genesisBlock.Coinbase())
 
 	// Simple transfer from signer to random account
-	startBalance, err := l2Seq.BalanceAt(context.Background(), fromAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
-	require.Nil(t, err)
+	startBalance := balanceAt(fromAddr)
```

Comment:
Thanks for introducing `balanceAt`! The initial balance calls were targeting block number `rpc.EarliestBlockNumber`, not `nil`. Maybe we change the `balanceAt` in this test to have a second `*big.Int` parameter so we can optionally specify the block number, and then add back `rpc.EarliestBlockNumber` as a parameter for the initial balances?

Reply:
Done